mevislab.github.io/content/tutorials/summary/summary8.md

```diff
@@ -0,0 +1,327 @@
+
+You can emulate the final viewer by adding a `RemoteRenderingClient` module to the outputs of your web macro.
+
+![RemoteRenderingClient](/images/tutorials/summary/Example8_9.png "RemoteRenderingClient")
+
+Open the *\*.script* files of your macro modules and copy the *FieldListeners* from *Commands* section of your *TutorialSummary.script* to *TutorialSummaryBrowser.script*.
+
+{{< highlight filename="TutorialSummaryBrowser.script" >}}
+```Stan
+Commands {
+  source = $(LOCAL)/TutorialSummaryBrowser.py
+  
+  FieldListener selected3DView {
+    command = viewSelectionChanged
+  }
+  FieldListener resetApplication {
+    command = resetApplication
+  }
+  FieldListener markerPosition {
+    command = insertPosition
+  }
+  FieldListener applyMarker {
+    command = applyPosition
+  }
+}
+```
+{{</highlight>}}
+
+Also copy the *Window* section to your web macro module. The *Box* of the *Viewing* tab needs to be modified because we are not using the `View3D` and `SoExaminerViewer` outputs anymore but the `RemoteRendering` outputs.
+
+{{< highlight filename="TutorialSummaryBrowser.script" >}}
+```Stan
+Window "MainPanel" {
+  // Define minimum width and height
+  minimumWidth  = 400
+  minimumHeight = 300
+  initCommand   = resetApplication
+  // Vertical Layout and 4 Boxes with Horizontal Layout
+  Vertical {
+    Box Source {
+      layout = Horizontal
+      Field openFile {
+        browseButton = True
+        browseMode   = open
+      }
+      Field resetApplication { }
+    }
+    Box Viewing {
+      layout = Horizontal
+      RemoteRendering out2D {
+        expandX = True
+        expandY = True
+      }
+      RemoteRendering out3D {
+        expandX = True
+        expandY = True
+      }
+    }
+    Box Settings {
+      layout = Horizontal
+      Field selectOverlayColor {
+        title   = Color
+      }
+      Field selectOverlayTransparency {
+        title   = Alpha
+      }
+      Field imageAlpha {
+        step   = 0.1
+        slider = True
+      }
+      Field thresholdInterval {
+        step   = 0.1
+        slider = True
+      }
+      Field isoValueImage {
+        step   = 2
+        slider = True
+      }
+      Field markerPosition {}
+      Field applyMarker {}
+      ComboBox selected3DView {
+        alignX   = Left
+        editable = False
+      }
+    }
+    Box Info {
+      layout    = Horizontal
+      Field totalVolume {}
+    }
+  }
+}
+```
+{{</highlight>}}
+
+#### Python functions
+After we re-used the scripts, we now need to copy the Python functions from *TutorialSummary.py* to *TutorialSummaryBrowser.py*. Open the Python file of your web macro. You will see an additional import from *MLABRemote*, which is required for remote rendering calls. The *MLABRemote* context is already setup automatically and can be used.
+
+{{< highlight filename="TutorialSummaryBrowser.py" >}}
+```Python
+from mevis import *
+
+from MLABRemote import *
+MLABRemote = setupRemoteContext(ctx)
```

Comment:
This looks strange - you first import everything from the module, and then redefine it? Maybe I'm missing something...

Reply:
this is the default in MeVisLab

Reply:
I have checked with Uwe, and that does not make much sense. Just use
```py
from MLABRemote import *
setupRemoteContext(ctx)
```

Reply:
Then we should also change the template in the software, because this is what is generated in MeVisLab. I change it but pls also change template.